src/prommis/leaching/leach_train.py

```diff
@@ -0,0 +1,315 @@
+r"""
+Leach Train
+===========
+
+Author: Andrew Lee
+
+The Leach Train unit model represents a series of leaching tanks in which a solid and liquid stream are contacted and
+undergo heterogeneous chemical reactions.
+
+Configuration Arguments
+-----------------------
+
+When creating an instance of a Leach Train model, the user must specify the number of tanks in train using the
+``number_of_tanks`` configuration argument. This value cannot be changed later.
+
+A Leach Train model also requires a special "heterogeneous reaction package" to define the reactions which occur in the
+system and the constraints defining the rate or reaction or equilibrium condition associated with each. These custom
+reaction packages are similar to normal IDAES reaction packages with the following differences:
+
+* the reaction package is not coupled to any specific thermophysical property package (i.e., the liquid or solid phase
+properties). Instead, it is expected that the reaction package will link to both the solid and liquid phase properties
+using ``self.parent_block().liquid`` and ``self.parent_block().solid`` as required.
+
+Degrees of Freedom
+------------------
```

Comment:
Shouldn't this also include liquid volumetric flow, liquid mass concentrations, solid mass flow, and solid component mass fracs?

Reply:
No - those are not degrees of freedom of the unit model, but are instead related to the the property packages. Note that if the LeachTrain is not the first unit in the flowsheet, the user would not need to fix these.